docs: features and env vars (#1815)
opendp/opendp: docs/source/contributing/development-environment.rst

In the above commands, the features ``untrusted`` and ``bindings`` are enabled.

Setting a feature changes how the crate compiles. 
In addition to the :ref:`feature-listing`, Rust also has the following features:

.. _rust-feature-listing:

.. dropdown:: Rust-Exclusive Feature List

    
   .. list-table::
      :widths: 25 75
      :header-rows: 1

>       * - ``untrusted``

Reviewer: ```suggestion
      * - Name
        - Description
      * - ``untrusted``
```
The first row is bolded.

I'd make this a full list, rather than just a subset. Slightly redundant, but good to have it all in one place.

        - Enables untrusted features ``contrib`` and ``floating-point``.
      * - ``ffi``
        - Enable to include C foreign function interfaces.
      * - ``derive``
        - Enable to support code generation and links to proofs in documentation.
      * - ``bindings``
        - Enable to generate Python and R source code. Also enables the ``ffi`` and ``derive`` features. 
      * - ``partials``
        - Enabled by default. When enabled, ``then_*`` functions are generated from ``make_*`` functions. Also enables the ``derive`` feature.
      * - ``use-openssl``
        - Enabled by default. Use OpenSSL for secure noise generation.

> To make the crate compile faster, FFI functions in debug builds support a reduced set of primitive types.

Reviewer: Is this still true? I know there was a recent change, but maybe there are still gaps, though we expect to hit them less often?

Author: The recent change withdraws this restriction only for two APIs out of all of our APIs (probably like .1%).

    tools/r_stage.sh -d

Environment Variables
---------------------

.. list-table::
   :widths: 25 75
   :header-rows: 1

   * - Name
     - Description
   * - ``OPENDP_LIB_DIR``

>      - Overrides the directory in which OpenDP language bindings look for the OpenDP Library binary.
>        It is recommended to set this when developing with the R bindings.

Reviewer: ```suggestion
     - Overrides the directory in which the OpenDP language binding looks for the OpenDP Library binary.
       See example above.
```
With a link to the `#r-setup` section?

   * - ``OPENDP_POLARS_LIB_PATH``
     - Each OpenDP Polars plugin contains a path to the OpenDP Library binary.
       When OpenDP is used as a query server, library paths in queries submitted by clients are stale (local to the client).

>        This environment variable overrides paths in private or stable OnceFrames emitted by the library.

Reviewer: ```suggestion
       This environment variable overrides paths in new OnceFrames.
```
Perhaps simplify, if it's still accurate?

>        For Python, you can find this path at ``opendp._lib.lib_path``.

Reviewer: If this is read-only, mention that?
```suggestion
       For Python, you can read this value from ``opendp._lib.lib_path``.
```

   * - ``OPENDP_HEADLESS``

>      - Used by CI. When ``true``, OpenDP will import without the presence of the OpenDP Library binary.

Reviewer: ```suggestion
     - Used by CI. When ``true``, The Python `opendp` package will import without the presence of the OpenDP Library binary.
```

   * - ``OPENDP_SPHINX_PORT`` and ``OPENDP_SPHINX_URI``
     - When configured, links to proof documents hosted by Sphinx point to the URI and port.
       The URI defaults to localhost. 
       Allows for a local documentation site.

>        Start the server from ``\docs`` with ``make sphinx-server``.

Reviewer: ```suggestion
       Start the server from ``docs/`` with ``make sphinx-server``.
```

   * - ``OPENDP_RUSTDOC_PORT`` and ``OPENDP_RUSTDOC_URI``

>      - When configured, links to proof documents to Rustdocs point to the URI and port.

Reviewer: ```suggestion
     - When configured, links in proof documents to Rustdocs include the URI and port.
```

       The URI defaults to localhost. 
       Allows for a local documentation site.

>        Start the server from ``\docs`` with ``make rustdoc-server``.

Reviewer: ```suggestion
       Start the server from ``docs/`` with ``make rustdoc-server``.
```
